source/_posts/2017-02-04-babyphone.markdown

```diff
@@ -0,0 +1,96 @@
+
+<p class='note'>
+We change the platform name for binary sensor in 0.38 from `ffmpeg` to `ffmpeg_noise`. Also all service going to component and was rename from `binary_sensor.ffmpeg_xy` to `ffmpeg.xy`.
+</p>
+
+On Raspbian Jessie, you can setup [ffmpeg](/components/ffmpeg) and install a [icecast2](http://icecast.org/) server using:
+```bash
+$ sudo echo "deb http://ftp.debian.org/debian jessie-backports main" >> /etc/apt/sources.list
+$ sudo apt-get update
+$ sudo apt-get -t jessie-backports install ffmpeg
+$ sudo apt-get install icecast2
+```
+
+We setup a icecast mount point for our babyphone and update `/etc/icecast2/icecast.xml`:
+```
+<mount>
+    <mount-name>/babyphone.mp3</mount-name>
+    <stream-name>Babyphone</stream-name>
+
+    <username>stream_user</username>
+    <password>stream_pw</password>
+</mount>
+```
+
+Now we can add the noise sensor to Home-Assistant. We can lower the sensitivity of the sensor (so that you are not inundated with notifications for every cough of the baby) to 2 seconds using the `duration` option. The sensor should wait 60 seconds before restoring and it prevent us that a wine break will triggering a new alarm.
+
+We can optimize the audio stream for human voice by using a highpass filter with 300Hz and a lowpass filter with 2500Hz. This filters out all non-human sounds such as background noise. We can even add a volume amplifier if the microphone volume is too low (you can remove it from `extra_arguments`). For icecast2 we convert the audio stream to mp3 with samplerate of 16000 (which is the minimum for Sonos speakers). We use `peak` to set the threshold for noise detection, where 0 dB is very loud and -100 dB is low.
+
+```yaml
+binary_sensor:
+ - platform: ffmpeg_noise
+   input: rtsp://user:pw@my_input/video
+   extra_arguments: -filter:a highpass=f=300,lowpass=f=2500,volume=volume=2 -codec:a libmp3lame -ar 16000
+   output: -f mp3 icecast://stream_user:stream_pw@127.0.0.1:8000/babyphone.mp3
+   initial_state: false
+   duration: 2
+   reset: 60
+   peak: -32
+```
+
+We use the option `initial_state` to prevent the ffmpeg process from starting with Home-Assistant and only start it when needed. We use an `input_boolean`  to control the state of ffmpeg services using the following automation.
+
+```
+input_boolean:
+  babyphone:
+    name: babyphone
+    initial: off
+
+automation:
+ - alias: 'Babyphone on'
+   trigger:
+     platform: state
+     entity_id: input_boolean.babyphone
+     from: 'off'
+     to: 'on'
+   action:
+     service: ffmpeg.start
+     entity_id: binary_sensor.ffmpeg_noise
+ - alias: 'Babyphone off'
+   trigger:
+     platform: state
+     entity_id: input_boolean.babyphone
+     from: 'on'
+     to: 'off'
+   action:
+     service: ffmpeg.stop
+     entity_id: binary_sensor.ffmpeg_noise
+```
+
+### {% linkable_title Thanks %}
+
+Special thanks for support to write this blogpost going to:
```

Comment:
Special thanks to [arsaboo](https://github.com/arsaboo) for assistance in writing this blogpost.